ANMN/burst_averaged_product/burst_average.py

```diff
@@ -0,0 +1,397 @@
+                                                                          time_max, datetime.now().strftime("%Y%m%dT%H%M%SZ"))
+
+    return burst_filename
+
+def create_burst_average_netcdf(input_netcdf_file_path, output_dir):
+    """
+    generate the burst netcdf file for WQM product.
+    see variable conf_file if editing of gatt and var att need to be done
+    """
+    input_netcdf_obj = Dataset(input_netcdf_file_path, 'r')
+    burst_vars       = create_burst_average_var(input_netcdf_obj)
+    time_burst_vals  = burst_vars['TEMP']['time_mean']
+    tmp_netcdf_dir   = tempfile.mkdtemp()
+
+    output_netcdf_file_path = os.path.join(tmp_netcdf_dir, generate_netcdf_burst_filename(input_netcdf_obj, burst_vars))
+    output_netcdf_obj       = Dataset(output_netcdf_file_path, "w", format="NETCDF4")
+
+    # read gatts from input, add them to output. Some gatts will be
+    # overwritten
+    input_gatts     = input_netcdf_obj.__dict__.keys()
+    gatt_to_dispose = ['author', 'toolbox_input_file', 'file_version_quality_control', 'quality_control_set',
+                       'history', 'CoordSysBuilder_', 'netcdf_filename', 'metadata', 'featureType']
+    for gatt in input_gatts:
+        if gatt not in gatt_to_dispose:
+            setattr(output_netcdf_obj, gatt, getattr(input_netcdf_obj, gatt))
+
+    if 'WQM' in output_netcdf_obj.instrument:
+        instrument_type = 'WQM'
+    elif 'CTD' in output_netcdf_obj.instrument:
+        instrument_type = 'CTD'
+
+    if instrument_type == 'WQM':
+        output_netcdf_obj.title = 'Burst-averaged biogeochemical measurements at %s' % (input_netcdf_obj.site_code)
+    elif instrument_type == 'CTD':
+        output_netcdf_obj.title = 'Burst-averaged moored CTD measurements at %s' % (input_netcdf_obj.site_code)
+
+    output_netcdf_obj.input_file   = get_input_file_rel_path(input_netcdf_file_path)
+    output_netcdf_obj.history      = 'Created %s' % time.ctime(time.time())
+    output_netcdf_obj.date_created = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    output_netcdf_obj.geospatial_lat_min = input_netcdf_obj['LATITUDE'][:].min()
+    output_netcdf_obj.geospatial_lat_max = input_netcdf_obj['LATITUDE'][:].max()
+    output_netcdf_obj.geospatial_lon_min = input_netcdf_obj['LONGITUDE'][:].min()
+    output_netcdf_obj.geospatial_lon_max = input_netcdf_obj['LONGITUDE'][:].max()
+
+    output_netcdf_obj.geospatial_vertical_min = min(burst_vars['DEPTH']['var_mean'])
+    output_netcdf_obj.geospatial_vertical_max = max(burst_vars['DEPTH']['var_mean'])
+
+    output_netcdf_obj.time_coverage_start = num2date(time_burst_vals, input_netcdf_obj['TIME'].units, input_netcdf_obj['TIME'].calendar).min().strftime('%Y-%m-%dT%H:%M:%SZ')
+    output_netcdf_obj.time_coverage_end   = num2date(time_burst_vals, input_netcdf_obj['TIME'].units, input_netcdf_obj['TIME'].calendar).max().strftime('%Y-%m-%dT%H:%M:%SZ')
```

Comment:
You could make this neater and more readable by doing the `num2date` just once and then using the result in the above 2 lines.